nucypher/blockchain/eth/actors.py

```diff
@@ -1,250 +1,310 @@
+        r = r.format(class_name, self.checksum_public_address)
         return r
 
+    @classmethod
+    def from_config(cls, config):
+        """Read actor data from a configuration file, and create an actor instance."""
+        raise NotImplementedError
+
+    @property
     def eth_balance(self):
         """Return this actors's current ETH balance"""
-        balance = self.token_agent.blockchain.interface.w3.eth.getBalance(self.address)
+        balance = self.token_agent.blockchain.interface.w3.eth.getBalance(self.checksum_public_address)
         return balance
 
+    @property
     def token_balance(self):
         """Return this actors's current token balance"""
-        balance = self.token_agent.get_balance(address=self.address)
+        balance = self.token_agent.get_balance(address=self.checksum_public_address)
         return balance
 
 
 class Miner(NucypherTokenActor):
     """
-    Ursula - practically carrying a pickaxe.
+    Ursula baseclass for blockchain operations, practically carrying a pickaxe.
     """
 
-    class StakingError(NucypherTokenActor.ActorError):
+    class MinerError(NucypherTokenActor.ActorError):
         pass
 
-    def __init__(self, miner_agent: MinerAgent=None, *args, **kwargs):
-        if miner_agent is None:
-            miner_agent = MinerAgent(token_agent=NucypherTokenAgent())
+    def __init__(self, is_me=True, miner_agent: MinerAgent = None, *args, **kwargs):
+        miner_agent = miner_agent if miner_agent is not None else MinerAgent()
         super().__init__(token_agent=miner_agent.token_agent, *args, **kwargs)
 
+        # Extrapolate dependencies
         self.miner_agent = miner_agent
-        miner_agent.miners.append(self)    # Track Miners
-
         self.token_agent = miner_agent.token_agent
         self.blockchain = self.token_agent.blockchain
 
-        self.__locked_tokens = None
-        self.__update_locked_tokens()
+        # Establish initial state
+        self.is_me = is_me
+
+    @classmethod
+    def from_config(cls, blockchain_config) -> 'Miner':
+        """Read miner data from a configuration file, and create an miner instance."""
 
-    def __update_locked_tokens(self) -> None:
-        """Query the contract for the amount of locked tokens on this miner's eth address and cache it"""
+        # Use BlockchainConfig to default to the first wallet address
+        wallet_address = blockchain_config.wallet_addresses[0]
 
-        self.__locked_tokens = self.miner_agent.contract.functions.getLockedTokens(self.address).call()
+        instance = cls(checksum_address=wallet_address)
+        return instance
 
+    #
+    # Staking
+    #
     @property
     def is_staking(self):
         """Checks if this Miner currently has locked tokens."""
-
-        self.__update_locked_tokens()
-        return bool(self.__locked_tokens > 0)
+        return bool(self.locked_tokens > 0)
 
     @property
-    def locked_tokens(self,):
+    def locked_tokens(self, ):
```

Comment:
What ahhh... what are we looking at here?

Reply:
hah